source/fundamentals/crud/read-operations/count.txt

```diff
@@ -0,0 +1,247 @@
+
+Modify Behavior
+~~~~~~~~~~~~~~~
+
+You can modify the behavior of ``CountDocuments()`` by passing a ``CountOptions`` type as 
+a parameter. If you don't specify any options, the driver uses default values.
+
+You can set the following properties in a ``CountOptions`` object:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Property
+     - Description
+
+   * - ``Collation``
+     - | The type of language collation to use when sorting results.  
+       | Default: ``nil``
+
+   * - ``Hint`` 
+     - | The index to use to scan for documents to count. 
+       | Default: ``nil``
+
+   * - ``Limit`` 
+     - | The maximum number of documents to count. 
+       | Default: ``0`` 
+
+   * - ``MaxTime``
+     - | The maximum amount of time that the query can run on the server.
+       | Default: ``nil``
+
+   * - ``Skip`` 
+     - | The number of documents to skip before counting.
+       | Default: ``0``
+
+.. tip::
+
+   When you use ``CountDocuments()`` to return the total number of documents in a
+   collection, MongoDB performs a collection scan. You can avoid a collection scan and
+   improve the performance of this method by using a hint to take advantage of the built-in index on
+   the ``_id`` field. Use this technique only when calling ``CountDocuments()``
+   with an empty query parameter.
+
+   .. code-block:: csharp
+      :emphasize-lines: 1, 2
+
+      var filter = Builders<Student>.Filter.Empty;
+      CountOptions opts = new CountOptions(){Hint = "_id_"};
+      var counter = collection.CountDocuments(filter, opts);
+
+.. _csharp-estimated-count:
+
+Estimated Count
+---------------
+
+To estimate the total number of documents in your collection, use the
+``EstimatedDocumentCount()`` method. 
+
+.. note:: 
+
+    The ``EstimatedDocumentCount()`` method is quicker than the
+    ``CountDocuments()`` method because it uses the collection's
+    metadata rather than scanning the entire collection. 
+
+Modify Behavior
+~~~~~~~~~~~~~~~
+
+You can modify the behavior of ``EstimatedDocumentCount()`` by passing a
+``EstimatedDocumentCountOptions`` type as a parameter. If you don't
+specify any options, the driver uses default values.
+
+You can set the following properties in a ``EstimatedDocumentCountOptions`` object:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Property
+     - Description
+
+   * - ``MaxTime``
+     - | The maximum amount of time that the query can run on the server.
+       | Default: ``nil``
+
+Example
+```````
+
+The following example estimates the number of documents in the
+``students`` collection:
+
+.. io-code-block::
+   :copyable: true
+
+   .. input::
+      :language: csharp
+      :dedent:
+
+     var count = _myColl.EstimatedDocumentCount();
+     Console.WriteLine("Estimated number of documents in the students collection: " + count);
+
+   .. output::
+      :language: none
+      :visible: false
+
+      Estimated number of documents in the students collection: 6
```

Comment:
I: example is not showing up in staging because of indentation errors
S: fix errors (text should be aligned with directive)